github3/repos/branch.py

```diff
@@ -401,6 +550,79 @@ def _update_attributes(self, protection):
             self,
         )
 
+    def set_required_code_owner_reviews(self, value):
+        """
+        Sets the mandatory requirement for code owner reviews.
+
+        :param bool value:
+            Do we require the code owner reviews.
+        """
+        self.require_code_owner_reviews = value
+
+    def set_dismiss_stale_reviews(self, value):
+        """
+        Do we dismiss the stale reviews once the code is updated?
+
+        :param bool value:
+            Should we dismiss the stale reviews once the code is updated?
+        """
+        self.dismiss_stale_reviews = value
+
+    def set_required_approving_review_count(self, value):
+        """
+        Sets how many approving reviews of the pull request are required
+        in order for the Pull Request to be able to get merged.
+
+        :param int value:
+            How many approving reviews are required.
+        """
+        self.required_approving_review_count = value
+
+    @decorators.requires_auth
+    def update(self):
+        """
+        Updates the configuration for the Required Pull Request Reviews.
+
+        :returns:
+            A updated instance of the required pull request reviews.
+        :rtype:
+            :class:`~github3.repos.branch.ProtectionRequiredPullRequestReviews`
+        """
+        teams = []
+        for team in self.dismissal_restrictions.original_teams:
+            teams.append(team.slug)
+
+        users = []
+        for user in self.dismissal_restrictions.original_users:
+            users.append(user.login)
+
+        update_json = {
+            'dismiss_stale_reviews': self.dismiss_stale_reviews,
+            'require_code_owner_reviews': self.require_code_owner_reviews,
+            'required_approving_review_count':
+                self.required_approving_review_count,
+            'dismissal_restrictions': {
+                'teams': teams,
+                'users': users
+            }
+        }
+        resp = self._patch(self._api, json=update_json)
+        json = self._json(resp, 200)
+        return ProtectionRequiredPullRequestReviews(json=json,
```

Comment:
@sigmavirus24 I need your help here. When executing the test for [update method](https://github.com/paulchubatyy/github3.py/blob/8e72b22d1085ba125706e1e08242a444eab8153c/tests/unit/test_repos_branch_protection.py#L104) I would expect the response to have the json as defined in [pull_request_reviews_example](https://github.com/paulchubatyy/github3.py/blob/8e72b22d1085ba125706e1e08242a444eab8153c/tests/unit/json/branch_protection_required_pull_request_reviews_example). Then the instance of the `ProtectionRequiredPullRequestReviews` class would be updated. Sadly it doesn't happen that way and I do not understand why. This [test output](https://travis-ci.org/sigmavirus24/github3.py/jobs/365259446#L484) might give a hint what I'm doing wrong here.

Reply:
You can use `self._instance_or_null` here for now. The project needs to figure out how to do our unit-testing better now that we likely won't return `None` from these methods in real-life.

I think the other problem here is that the `unit` tests don't talk to the internet and return a `None` response, effectively because all we care about is verifying that we make the request correctly.

Reply:
Would it be a good idea to tweak the unit tests so that they didn't go to the internet but grabbed the example responses from the same source we use for initialization?

Anyway, I will update my code to follow your suggestion.

Reply:
> Would it be a good idea to tweak the unit tests so that they didn't go to the internet but grabbed the example responses from the same source we use for initialization?

I don't think so. Based on what we care about, we don't need them to return something and setting that up would be a lot more work than we can currently do and would apply to *all* of our tests. It would be a lot of work to retrofit that.